web/server/codechecker_server/api/report_server.py

```diff
@@ -947,6 +1003,13 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
 
                 q = q.limit(limit).offset(offset)
 
+                report_details = {}
+
+                # Get report details if it is required.
+                if get_details:
+                    report_ids = [r[1] for r in q]
+                    report_details = self.__get_report_details(report_ids)
```

Comment:
Couldn't we we use the already created session to the report details?

If getting details were done in the for loop below we could get the details for the unique results too right?

Reply:
We are getting uniqued reports in a different way, see the code above:
https://github.com/Ericsson/codechecker/blob/303cac9af72527a107cbdad8b3526a09ef1a252d/web/server/codechecker_server/api/report_server.py#L907-L921

I think for uniqued reports it doesn't make any sense to get the details.

Reply:
I still do not understand why it does not make sense to get the report details for the unique reports?
It is possible that I want to know the bugpath for a unique report.

https://github.com/Ericsson/codechecker/blob/52cbd0fc7ef7a3109e5540cd3c1c46ed48d1dbc7/web/server/codechecker_server/api/report_server.py#L907-L921

Calling `report_details = get_report_details(session, [report_id])` to get the report details in the loop could be done for the unique reports too.